### PR TITLE
1901/program questions application

### PR DESCRIPTION
**Pull request by dominikx96**
# Pull Request Template

## Issue Overview

This PR addresses #1901

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Adds program questions section into the application management form.

## How Can This Be Tested/Reviewed?

1. Create or use existing listing with defined program questions
2. Add a new application and try multiple program answers combination
3. Make sure that data has been saved correctly
4. Try to edit an application
5. Make sure that edit progress saved proper values

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by jaredcwhite** [comment]
@dominikx96 OK, so I was able to wrap up the latest changes to #2114, submit an application, then look at the review app for this PR and see that the data was displayed on the partners side. Yay! The labels aren't showing up quite right, but I think maybe copying in the latest data I have in `general.json` translation file would fix it. Example:

image

File: sites/partners/src/applications/PaperApplicationForm/sections/FormPrograms.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
+import React, { useMemo } from "react"
```
Comment (emilyjablonski) [inline sites/partners/src/applications/PaperApplicationForm/sections/FormPrograms.tsx:1]: There's a couple variables in here that are named with `preference` that don't look intentional, can you check and change to program?

**Review by YazeedLoonat — CHANGES_REQUESTED** [review]
Looks good, just a small code change request

**Comment by dominikx96** [comment]
Closed, newest version: https://github.com/bloom-housing/bloom/pull/2182

**Pull request closed without merging**